src/topkg.mli

```diff
@@ -754,6 +754,13 @@ module Conf : sig
       [pinned c] is the value of a predefined key [--pinned]. It has
       exactly the same semantics as {!dev_pkg} but is misnamed. *)
 
+  val jobs : t -> int
+  (** [jobs c] is the value of a predefined key [--jobs].
+      If absent it is determined from the build context as follows.
+      {ul
+      {- [`Dev] builds default to number of CPU cores, or 4 if it cannot be determined.}
+      {- all other contexts default to 4}} *)
```

Comment:
Due to the lack of a jobserver (like `make` has) during an `opam install` you want to avoid spawning too many jobs (i.e. `JOBS` * `OPAM_JOBS`). This can be a problem on a machine with many cores (e.g. 32). `rpmbuild` has a builting default limit in its macros of 16 too.
I think the limit should be smaller than 8, considering a typical developer's machine is unlikely to have more than 8 cores, and can do a few opam installs in parallel.
`4` is just an arbitrary small number, that happens to match the jbuilder default.

For `Dev builds we know it is a build run by the developer, and not from an `opam` context, so we can safely use all the CPUs.

If the logic is too complicated I'd be happy to replace it with a default of `max 16 ncpus` instead of the logic depending on `Dev.